### PR TITLE
issue 915 to refactor helm-update-config code

### PR DESCRIPTION
**Pull request by EmilyEmily**

### CODE REVIEW

**Comment by EmilyEmily** [comment]
```
helm update-config "cf.20190214.142635" --set-value=sizing.diego_cell.count="1",kube.diego_cell_size="1"
```
works well with this version of plugin.

File: Makefile [hunk 1]
```diff
@@ -1,5 +1,5 @@
 build:
-	@gox -osarch "darwin/amd64 linux/amd64" -output "dist/{{.Dir}}_{{.OS}}_{{.Arch}}"
+	@gox -osarch "darwin/amd64 linux/amd64" -output "dist/{{.Dir}}_{{.OS}}_{{.Arch}}" helm-update-config
 
 release:
 	@ghr -t ${GITHUB_TOKEN} -u burdiyan -r helm-update-config --replace `git describe --tags` dist/
```
Comment (zhanggbj) [inline Makefile]: Hi @EmilyEmily ,

Would you please help update the release and the readme doc, including use name(burdiyan), install guide and usage, as we'll maintain our own repo in future. Thanks!

**Review by mattcui — COMMENTED** [review]
updatecfg.go should be in a separated dir like `cmd`?

File: updatecfg_test.go [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+		values[0] = "--set-value=foo.bar=\"9\""
+		uValues, err := updatec.GenerateUpdatedValues(vf, values)
+
+		fmt.Println(uValues)
+		fmt.Println("error:", err)
+		Expect(true)
+		//		Expect(values[0].To(Equal("--set-value=foo.bar=\"9\"")))
+		//Expect(uValues["test"].To(Equal(1)))
+		//		Expect(uValues["foo"]["baz"].To(Equal(6)))
+		//		Expect(uValues["qux"]["uier"].To(Equal(false)))
```
Comment (zhanggbj) [inline updatecfg_test.go]: Hi @EmilyEmily,

Looks like hese lines are commented. I think if they are not used, it's better to remove them. What do you think? Thanks!

File: Makefile (hunk 1 above)
Comment (EmilyEmily) [reply Makefile]: Thanks for the reminder, updated releases and readme doc. 
I'm working on ginkgo unit test now, and will push a new version later.

File: README.md [hunk 3]
```diff
@@ -5,27 +5,29 @@ This plugin adds `helm update-config` command to Helm CLI. It allows to update c
...
 ## Usage
 
 To change image tag of `smiling-penguin` release:
 
 ```
-helm update-config smiling-penguin --set=image.tag=stable
+helm update-config smiling-penguin --set-value=image.tag=stable
+helm update-config smiling-penguin --reset-value=true
+helm update-config smiling-penguin --value=file_path1
 ```
```
Comment (zhanggbj) [inline README.md:19]: Hi @EmilyEmily,

As we also support both --set-value and --value together, would you please also add this in usage, and we prefer the value in --set-value if there's a same one in --value file. Thank you!

**Review by zhanggbj — CHANGES_REQUESTED** [review]

**Review by mattcui — COMMENTED** [review]
Maybe reference helm plugin projects in golang, like https://github.com/ContainerSolutions/helm-convert

**Review by mattcui — COMMENTED** [review]
Go allows us to specify the books_test package alongside the books package. Using books_test instead of books allows us to respect the encapsulation of the books package: your tests will need to import books and access it from the outside, like any other package. This is preferred to reaching into the package and testing its internals and leads to more behavioral tests. **You can, of course, opt out of this – just change package books_test to package books**

I think for this case, we should set the same package for test and then you can test all private functions, which is what we expect here.

**Comment by EmilyEmily** [comment]
> I think for this case, we should set the same package for test and then you can test all private functions, which is what we expect here.

I communicated with other team, they mentioned that normally they only test public functions, and during testing, private functions will be called by public functions internally, thus private functions will be tested indirectly.

File: Makefile (hunk 1 above)
Comment (EmilyEmily) [reply Makefile]: done

File: updatecfg_test.go (hunk 2 above)
Comment (EmilyEmily) [reply updatecfg_test.go]: done

File: README.md (hunk 3 above)
Comment (EmilyEmily) [reply README.md:19]: added samples.

File: .travis.yml [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+- go get github.com/onsi/gomega
+- go get github.com/pkg/errors
+- go get github.com/modocache/gover
+- go get github.com/mattn/goveralls
+- go get golang.org/x/tools/cmd/cover
+install: true
+script: bin/unit_test
+branches:
+  only:
+  - emilyhelmupdate
```
Comment (zhanggbj) [inline .travis.yml]: Hi,@EmilyEmily 

After merge this PR, travis will use `master` branch, right?

**Review by zhanggbj — CHANGES_REQUESTED** [review]

**Pull request merged**